airflow/providers/sftp/operators/sftp_batch.py

```diff
@@ -0,0 +1,294 @@
+                Possible options for PUT:
+                    1.optional(regexp_mask:str) + local_folder:str + remote_folder:str
+                    2.local_files_path:list + remote_folder:str
+                Possible options for GET:
+                    1.local_folder:str + remote_folder:str + optional(regexp_mask:str)
+                    2.local_folder:str + remote_files_path:list
+                """
+            )
+
+    def execute(self, context: Any) -> str:
+        dump_file_name_for_log = None
+        try:
+            _check_conn(self)
+
+            with self.ssh_hook.get_conn() as ssh_client:
+                sftp_client = ssh_client.open_sftp()
+                if self.operation.lower() == SFTPOperation.PUT:
+                    if self.local_folder and self.remote_folder:
+                        files_list = self._search_files(os.listdir(self.local_folder))
+                        for file in files_list:
+                            local_file = os.path.basename(file)
+                            dump_file_name_for_log = file
+                            self._check_remote_file(f"{self.remote_folder}/{local_file}", sftp_client)
+                            self._transfer(sftp_client, self.local_folder, local_file, self.remote_folder)
+                    if self.local_files_path and self.remote_folder:
+                        for file in self.local_files_path:
+                            local_file = os.path.basename(file)
+                            dump_file_name_for_log = file
+                            self._check_remote_file(f"{self.remote_folder}/{local_file}", sftp_client)
+                            self._transfer(sftp_client, os.path.dirname(file), local_file, self.remote_folder)
+                elif self.operation.lower() == SFTPOperation.GET:
+                    if self.remote_folder and self.local_folder:
+                        files_list = self._search_files(sftp_client.listdir(self.remote_folder))
+                        for file in files_list:
+                            remote_file = os.path.basename(file)
+                            dump_file_name_for_log = file
+                            self._check_local_file(f"{self.local_folder}/{remote_file}")
+                            self._transfer(sftp_client, self.local_folder, remote_file, self.remote_folder)
+                    if self.remote_files_path and self.local_folder:
+                        for file in self.remote_files_path:
+                            remote_file = os.path.basename(file)
+                            dump_file_name_for_log = file
+                            self._check_local_file(f"{self.local_folder}/{remote_file}")
+                            self._transfer(sftp_client, self.local_folder, remote_file, os.path.dirname(file))
+
+        except Exception as e:
+            raise AirflowException(f"Error while transferring {dump_file_name_for_log}, error: {str(e)}")
+
+        return self.local_folder
```

Comment:
Should this return `self.local_folder` and/or `self.remote_folder` depending on what is provided?

Reply:
its not a important, we may drop this line

Reply:
done